CMakeLists.txt

```diff
@@ -18,14 +18,23 @@ set(
   "1.ros_version"
   "1.ros_python_version"
 )
+set(shells "dsv")
 if(CMAKE_HOST_UNIX)
-  set(shell "sh")
+  list(APPEND shells "sh")
 else()
-  set(shell "bat")
+  list(APPEND shells "bat")
 endif()
 foreach(hook ${hooks})
-  ament_environment_hooks(
-    "${CMAKE_CURRENT_SOURCE_DIR}/env-hooks/${hook}.${shell}.in")
+  foreach(shell ${shells})
+    ament_environment_hooks(
+      "${CMAKE_CURRENT_SOURCE_DIR}/env-hooks/${hook}.${shell}.in")
+    ament_environment_hooks(
+      "${CMAKE_CURRENT_SOURCE_DIR}/env-hooks/${hook}.dsv.in")
```

Comment:
Since "dsv" is in the list `shells`, looks like we're calling the hooks command with the dsv hook twice. I think we can remove this command.

Reply:
Good catch. That is certainly a left over from previous iterations. Fixed in 84d1705ebd4c85c215013114967fc7da8b6cac11.